src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -86,9 +86,11 @@ export function DiscoverMainRoute({
     });
   const [error, setError] = useState<Error>();
   const [loading, setLoading] = useState(true);
-  const [hasESData, setHasESData] = useState(false);
-  const [hasUserDataView, setHasUserDataView] = useState(false);
-  const [showNoDataPage, setShowNoDataPage] = useState<boolean>(false);
+  const [noDataState, setNoDataState] = useState({
+    hasESData: true,
+    hasUserDataView: true,
```

Comment:
```suggestion
    hasESData: false,
    hasUserDataView: false,
```

Reply:
Or was the change intensional?

Reply:
No, it makes sense, shouldn't be relevant how this works, better this way